lib/subscription/batcher.ex

```diff
@@ -1,14 +1,37 @@
 defmodule AshGraphql.Subscription.Batcher do
+  @moduledoc """
+  If started as a GenServer, this module will batch notifications and send them in bulk.
+  Otherwise, it will send them immediately.
+
+  The Batcher can be configured during compile time using the following settings:
+
+  ```elixir
+  config :ash_graphql, :subscription_batch_settings,
+    # if there are more than 100 `async_limit` notifications, we will start to backpressure
+    async_limit: 100,
+    # if there are less then 50 `send_immediately_threshold` notifications, we will send them immediately
+    send_immediately_threshold: 50,
+    # the interval in milliseconds the batcher waits for new notifications before sending them
+    subscription_batch_interval: 1000
+  ```
+  """
   use GenServer
 
   alias Absinthe.Pipeline.BatchResolver
 
   require Logger
   @compile {:inline, simulate_slowness: 0}
 
-  defstruct batches: %{}, total_count: 0, async_limit: 100, send_immediately_threshold: 50
+  settings = Application.compile_env(:ash_graphql, :subscription_batch_settings, [])
+
+  defstruct batches: %{},
+            total_count: 0,
+            async_limit: Keyword.get(settings, :async_limit, 100),
+            send_immediately_threshold: Keyword.get(settings, :send_immediately_threshold, 50),
+            subscription_batch_interval: Keyword.get(settings, :subscription_batch_interval, 1000)
```

Comment:
`subscription_batch_interval` should be fetched from the `opts` on `init` as well. The recommended way to configure things like GenServers is through initialization options instead of compile time options. We can have both if we like, but the runtime configuration is generally better.

The way that it fetches the opts in `init` right now makes this work

```elixir
{Batcher, async_limit: 100}
```

And if we want them to configure it in config files, they should configure the module,

i.e `config :ash_graphql, AshGraphql.Subscription.Batcher, ...`

Reply:
Make sense. I'll remove the compile time again. You can add those options in `application.ex` then. 

But it's ok to also add the `subscription_batch_interval` there? That `TODO` comment was what made me doing this in the first place.

Reply:
Yeah, we should have that too 👍 I just forgot 😆 Thank you for catching that